### ISSUE TITLE
@apply with 'no-browser-hacks' will produce an Error

### ISSUE THREAD

**Issue by jyane** [stylelint/stylelint#2145]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?
Is this a bug?

> Which rule, if any, is this issue related to?

`no-browser-hacks`

> What CSS is needed to reproduce this issue?

```css
:root {
  --base: {
    color: #fff;
 }
}

.foo {
  @apply --base;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "no-browser-hacks": true
  }
}
```

> Which version of stylelint are you using?

`7.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

stylelint .

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

PostCSS

> What did you expect to happen?

No warnings, and No Errors.

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
Error: Expected pseudo-class or pseudo-element
    at new error (/.../jyane/temp/reproduce/node_modules/postcss-selector-parser/dist/processor.js:29:23)
```

**Comment by evilebottnawi** [comment]
@jyane `no-browser-hacks` rule used https://github.com/ben-eb/stylehacks package, this issue should be sent to this package
